src/sdk/lsf-sdk.js

```diff
@@ -218,19 +218,56 @@ export class LSFWrapper {
   }
 
   /** @private */
-  async loadTask(taskID, annotationID, fromHistory = false) {
+  async loadTask(taskID, annotationID, fromHistory = false, isPrevious = false) {
     if (!this.lsf) {
       return console.error("Make sure that LSF was properly initialized");
     }
 
     const nextAction = async () => {
       const tasks = this.datamanager.store.taskStore;
 
+      const projectId = this.datamanager.store.project.id;
+      const taskId = tasks.selectedId;
+      const annotationId = this.store.LSF.lsf.annotationStore.selected?.pk;
+      const draftId = this.store.LSF.lsf.annotationStore.selected?.draftId;
+      let _annotationId = annotationID;
+      let _taskId = taskID;
+
+      if (isPrevious && !isDefined(_taskId)) {
+        const props = {
+          projectId,
+        };
+
+        if (!this.lsf.noTask) {
+          if (taskId) {
+            props.task = taskId;
+          }
+
+          if (annotationId) {
+            props.annotation = annotationId;
+          } else if (draftId) {
+            props.draft = draftId;
+          }
+        }
+
+        const taskHistory = await tasks.loadTaskHistory(props);
+
+        if (!taskHistory[0].previous) {
+          this.store.LSF.lsf.task?.setHasPrevTask(false);
+          return;
+        } else {
+          this.store.LSF.lsf.task?.setHasPrevTask(true);
+        }
+
+        _taskId = taskHistory[0].previous.task;
+        _annotationId = taskHistory[0].previous.annotation || taskHistory[0].previous.draft;
+      }
+
       const newTask = await this.withinLoadingState(async () => {
-        if (!isDefined(taskID)) {
+        if (!isDefined(_taskId) && !isPrevious) {
           return tasks.loadNextTask();
         } else {
-          return tasks.loadTask(taskID);
+          return tasks.loadTask(_taskId);
```

Comment:
can we swap this for the affirmative so it reads a bit easier:

```suggestion
        if (isDefined(_taskId) && isPrevious) {
          return tasks.loadTask(_taskId);
        } else {
          return tasks.loadNextTask();
```